app/src/main/java/org/schabi/newpipe/database/stream/dao/StreamStateDAO.java

```diff
@@ -17,31 +17,31 @@
 import static org.schabi.newpipe.database.stream.model.StreamStateEntity.STREAM_STATE_TABLE;
 
 @Dao
-public abstract class StreamStateDAO implements BasicDAO<StreamStateEntity> {
+public interface StreamStateDAO extends BasicDAO<StreamStateEntity> {
     @Override
     @Query("SELECT * FROM " + STREAM_STATE_TABLE)
-    public abstract Flowable<List<StreamStateEntity>> getAll();
+    Flowable<List<StreamStateEntity>> getAll();
 
     @Override
     @Query("DELETE FROM " + STREAM_STATE_TABLE)
-    public abstract int deleteAll();
+    int deleteAll();
 
     @Override
-    public Flowable<List<StreamStateEntity>> listByService(final int serviceId) {
+    default Flowable<List<StreamStateEntity>> listByService(final int serviceId) {
         throw new UnsupportedOperationException();
     }
 
     @Query("SELECT * FROM " + STREAM_STATE_TABLE + " WHERE " + JOIN_STREAM_ID + " = :streamId")
-    public abstract Flowable<List<StreamStateEntity>> getState(long streamId);
+    Flowable<List<StreamStateEntity>> getState(long streamId);
 
     @Query("DELETE FROM " + STREAM_STATE_TABLE + " WHERE " + JOIN_STREAM_ID + " = :streamId")
-    public abstract int deleteState(long streamId);
+    int deleteState(long streamId);
 
     @Insert(onConflict = OnConflictStrategy.IGNORE)
-    abstract void silentInsertInternal(StreamStateEntity streamState);
+    void silentInsertInternal(StreamStateEntity streamState);
```

Comment:
The only issue I see with this change is that "internal" methods like this one have to be public (all of an interface's members can be _only_ public).

Reply:
Private interface methods were introduced in Java 9.

Reply:
@Isira-Seneviratne indeed, you're right; I was not aware of that. However, we're using Java 8 here, so we still can't do that.